src/DataStreams/PushingToViewsBlockOutputStream.cpp

```diff
@@ -40,10 +40,20 @@ PushingToViewsBlockOutputStream::PushingToViewsBlockOutputStream(
     /// We need special context for materialized views insertions
     if (!dependencies.empty())
     {
-        views_context = std::make_unique<Context>(context);
+        select_context = std::make_unique<Context>(context);
+        insert_context = std::make_unique<Context>(context);
+
+        const auto & insert_settings = insert_context->getSettingsRef();
+
         // Do not deduplicate insertions into MV if the main insertion is Ok
         if (disable_deduplication_for_children)
-            views_context->setSetting("insert_deduplicate", false);
+            insert_context->setSetting("insert_deduplicate", false);
+
+        // Separate min_insert_block_size_rows/min_insert_block_size_bytes for children
+        if (insert_settings.min_insert_block_size_rows_for_materialized_views.changed)
```

Comment:
suprising :) default is 0 but setting to zero explicitly change the behavior :)

Reply:
Yep, since it looks at `changed`, it was the intention, that way you can disable squashing for MV completely.

Reply:
Counter-intuitive for users. Better to have some different value for default (like -1)

Reply:
>Counter-intuitive for users

I guess you are right.

Anyway this place had been changed in #14005 - https://github.com/ClickHouse/ClickHouse/pull/14005/files#diff-54964bcc78d995d450f08be9f1a9e260e764ea784d6395dd7281c0f526be4a0eR61

And now `min_insert_block_size_rows_for_materialized_views=0` will not modify settings, regardless it had been changed or not.

If you want to disable squashing you should set `min_insert_block_size_rows_for_materialized_views=1`